include/gsl/span

```diff
@@ -291,20 +318,20 @@ namespace details
         constexpr void _Seek_to(const pointer p) noexcept
         { // adjust the position of *this to previously verified location p
             // after _Unwrapped
-            index_ = p - span_->data();
+            current_ = p;
         }
 #endif
 
-    protected:
-        const Span* span_ = nullptr;
-        std::ptrdiff_t index_ = 0;
+        pointer begin_ = nullptr;
+        pointer end_ = nullptr;
+        pointer current_ = nullptr;
     };
 
-    template <std::ptrdiff_t Ext>
+    template <std::size_t Ext>
     class extent_type
     {
     public:
-        using index_type = std::ptrdiff_t;
+        using index_type = std::size_t;
 
         static_assert(Ext >= 0, "A fixed-size span must be >= 0 in size.");
```

Comment:
Not pre-existing: This `static_assert` is now always true, and should instead be checking for `dynamic_extent`. Or it should be removed entirely, since there is an explicit specialization immediately below that will absorb `dynamic_extent`.